Documentation/api-guidelines/System.Memory.md

```diff
@@ -0,0 +1,46 @@
+# System.Memory Design Guidelines
+
+`System.Memory` is a collection of types and features that make working with
+buffers and raw memory more efficient while remaining type safe. The feature
+specs can be found here:
+
+* [`Span<T>`](https://github.com/dotnet/corefxlab/blob/master/docs/specs/span.md)
+* [`Memory<T>`](https://github.com/dotnet/corefxlab/blob/master/docs/specs/memory.md)
+
+## Overview
+
+* `ReadOnlySpan<T>` is effectively the universal receiver, in that `T[]`, `T*`,
+  `Memory<T>`, `ReadOnlyMemory<T>`, `Span<T>`, `ArraySegment<T>` can all be
+  converted to it.  So if you can declare your API to accept a `ReadOnlySpan<T>`
```

Comment:
Mention `string` can be converted to `ReadOnlySpan<char>`? Does there need to be any specific guidance on when to use `ReadOnlySpan<char>` vs. `string`?

Reply:
> Does there need to be any specific guidance on when to use ReadOnlySpan<char> vs. string?

Possibly. In general we should prefer `ReadOnlySpan<char>` when possible given it's a more universal receiver.

Reply:
This design still needs to close and is tracked as #25085.